helm/config_test.go

```diff
@@ -0,0 +1,800 @@
+		}
+	}
+
+	obj.Add("Very", NewScalar("Long", Comment(strings.Repeat(strings.Repeat("x", 50)+" ", 2))))
+	root.Add("Very", NewScalar("Long", Comment(strings.Repeat(strings.Repeat("x", 50)+" ", 2))))
+	root.Add("Nested", obj)
+
+	expect := `---
+# 12 12 12 12 12 12 12
+# 12 12 12
+Key2: ~
+# 123 123 123 123 123
+# 123 123 123 123 123
+Key3: ~
+# 1234 1234 1234 1234
+# 1234 1234 1234 1234
+# 1234 1234
+Key4: ~
+# 12345 12345 12345
+# 12345 12345 12345
+# 12345 12345 12345
+# 12345
+Key5: ~
+# 123456 123456 123456
+# 123456 123456 123456
+# 123456 123456 123456
+# 123456
+Key6: ~
+# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+Very: Long
+Nested:
+          # 12 12 12 12
+          # 12
+          Key2: ~
+          # 123 123 123
+          # 123 123
+          Key3: ~
+          # 1234 1234
+          # 1234 1234
+          # 1234
+          Key4: ~
+          # xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+          # xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+          Very: Long
+`
+	equal(t, root, expect, Indent(10), Wrap(24))
+}
+
+func TestHelmIndent(t *testing.T) {
+	obj1 := NewObject()
+	obj1.Add("Foo", NewScalar("Bar", Comment("Baz")))
+
+	list1 := NewList()
+	list1.Add(obj1)
+	list1.Add(NewScalar("1"))
+
+	list2 := NewList()
+	list2.Add(NewScalar("abc"))
+	list2.Add(NewScalar("xyz"))
+
+	list1.Add(list2)
+	list1.Add(NewScalar("2"))
+	list1.Add(NewScalar("3"))
+
+	obj2 := NewObject()
+	obj2.Add("List", list1)
+
+	obj3 := NewObject()
+	obj3.Add("Foo", NewScalar("1"))
+	obj3.Add("Bar", NewScalar("2"))
+
+	obj2.Add("Meta", obj3)
+
+	root := NewObject()
+	root.Add("Object", obj2)
+
+	expect := `---
+Object:
+    List:
+      - # Baz
+        Foo: Bar
+      - 1
+      -   - abc
+          - xyz
+      - 2
+      - 3
+    Meta:
+        Foo: 1
+        Bar: 2
+`
+	equal(t, root, expect, Indent(4))
+}
+
+func TestHelmEncoderModifier(t *testing.T) {
+	obj := NewObject()
+	obj.Add("foo", NewScalar("1"))
+	obj.Add("bar", NewScalar("2"))
+	obj.Add("baz", NewScalar("3"))
+
+	root := NewObject()
+	root.Add("Object", obj)
+
+	expect := `---
+Object:
+  foo: 1
+  bar: 2
+  baz: 3
+---
+Object:
+    foo: 1
+    bar: 2
+    baz: 3
+`
+
+	buffer := &bytes.Buffer{}
+	enc := NewEncoder(buffer, Indent(0))
+	enc.Encode(root)
+	enc.Set(Indent(4))
+	enc.Encode(root)
+	assert.Equal(t, expect, buffer.String())
+}
+
+func TestHelmMultiLineScalar(t *testing.T) {
+	root := NewObject()
+	root.Add("Scalar", NewScalar("foo\nbar\nbaz"))
+
+	list := NewList()
+	list.Add(NewScalar("one\ntwo\nthree"))
+	root.Add("List", list)
+
+	expect := `---
+Scalar: |-
+    foo
+    bar
+    baz
+List:
+  - |-
+    one
+    two
+    three
+`
+	equal(t, root, expect, Indent(4))
+}
+
+func TestHelmEmptyLines(t *testing.T) {
+	list := NewList()
+	list.Add(NewScalar("1", Comment("Some comment")))
+	list.Add(NewScalar("2"))
+	list.Add(NewScalar("3", Comment("Another comment")))
+	list.Add(NewScalar("4"))
```

Comment:
Might want to explicitly test `Comment("")` too (which shouldn't do anything).

Reply:
Done in e343ccdcc1574e9e4917fd4c6647ef36584ff08e. It doesn't change the output, as expected, as the empty string is the default value and used to check if the field is set or not.